web_environment_ribbon/static/src/js/ribbon.js

```diff
@@ -1,35 +1,57 @@
-/******************************************************************************
-    Copyright (C) 2015 Akretion (http://www.akretion.com)
-    @author Sylvain Calador <sylvain.calador@akretion.com>
+/* Copyright 2015 Sylvain Calador <sylvain.calador@akretion.com>
+   Copyright 2015 Javi Melendez <javi.melendez@algios.com>
+   Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
 
-    This program is free software: you can redistribute it and/or modify
-    it under the terms of the GNU Affero General Public License as
-    published by the Free Software Foundation, either version 3 of the
-    License, or (at your option) any later version.
+(function ($) {
```

Comment:
This must be an edge case due to no real dependencies then. 

I don't guess I'll block over this, but I think we may be setting a bad precedence by allowing old JS API in. I see zero examples of this old API in Odoo v9 code; they are all using the new style.

I'm still learning the true injection point for an app that should fire JS when the DOM is ready, but from what I understand the code should look similar to:

```
odoo.define('web_environment_ribbon.ribbon', function(require){
  "use strict";
  
  var $ = require('$');
  var Model = require('web.Model');
  var core = require('web.core');
  
  var model = new Model('ir.config_parameter');
  
  function validStrColour(strToTest) { ... }
  
  core.bus.on('web_client_ready', null, function () { Code to run here })
});
```